provider/pkg/provider/remote/connection.go

```diff
@@ -109,7 +114,9 @@ func (con *Connection) Dial(ctx p.Context, config *ssh.ClientConfig) (*ssh.Clien
 				net.JoinHostPort(*con.Host, fmt.Sprintf("%.0f", *con.Port)),
 				config)
 			if err != nil {
-				if try > 10 {
+				// on each try we already made a dial
+				dials := try + 1
+				if *con.DialErrorLimit > UnlimitedDialErrors && dials > *con.DialErrorLimit {
```

Comment:
Guessing we're counting on *deref not to panic because there's a default value specified.

Reply:
yeah well I assume that, just followed same approach as for [`port`](https://github.com/pulumi/pulumi-command/blob/694b12206afdcb529e92d72f30774c3524a5d76f/provider/pkg/provider/remote/connection.go#L109)